### ISSUE TITLE
 Error of feature's  extracking

### ISSUE THREAD

**Issue by tianboguangding** [AlfredXiangWu/face_verification_experiment#33]
Thanks for your shared model！

My problem is shape error，same with #9.
When I extract  the feature （‘fc1’），your code is ：

```
......
shp=blobs['fc1'].shape
......
features_shape = (len(image_list), shp[1], shp[2], shp[3]).
......
```

The layer's shape size is also 2,and info of shp is only :[20,512]. There is no info of shp[2],shp[3]

I try to change  test code for 6000 pairs matching:

```

...
import sklearn.metrics.pairwise as pw
def read_imagelist(filelist):
       fid =open(filelist)
       lines=fid.readlines()
       test_num=len(lines)
       fid.close()

       X=np.empty((test_num,1,128,128))
       i=0
       for line in lines:
              word=line.split('\n')
              filename=word[0]
              im1=skimage.io.imread(filename,as_gray=False)
              image=skimage.transform.resize(im1,(128,128))*1#Normalization

              #print 'gray:'+filename
              X[i,:,:]=image[:,:]

              i=i+1
       return X

def read_labels(labelfile):
       fin=open(labelfile)
       lines=fin.readlines()
       labels=np.empty((len(lines),))
       k=0;
       for line in lines:
              labels[k]=int(line)
              k=k+1;
       fin.close()
       return labels

def draw_roc(fpr,tpr,title='cosine',save_name='roc_lfw'):
       plt.figure()
       plt.plot(fpr,tpr)
       plt.plot([0,1],[0,1],'k--')
       plt.xlim([0.0,1.0])
       plt.ylim([0.0,1.0])
       plt.xlabel('FP')
       plt.ylabel('TP')
       plt.title('....:'+title)
       plt.legend(loc="lower right")
       plt.show()
       plt.savefig(save_name+'.png')


def evaluate(itera=500000,metric='cosine'):
       filelist_left='../left_all.txt'
       filelist_right='../right_all.txt'
       filelist_label='../label.txt'

       caffe.set_mode_gpu()
       net=caffe.Classifier('../DeepFace_set003_match.prototxt','../DeepFace_set003_net_iter.caffemodel')
       print 'net input:',net.inputs
       print 'net output:',net.outputs


       #Left feature
       X=read_imagelist(filelist_left)
       test_num=np.shape(X)[0]
       #out=net.forward_all(data=np.asarray([trans.preprocess('data',X)]))
       out=net.forward_all(data=X)
       ftr1=np.float64(out['fc1'])
       ftr1=np.reshape(ftr1,(test_num,512))

       #Right feature
       Y=read_imagelist(filelist_right)
       out2=net.forward_all(data=Y)
       ftr2=np.float64(out['fc1'])
       ftr2=np.reshape(ftr2,(test_num,512))

       labels=read_labels(filelist_label)
       assert(len(labels)==test_num)

       # cosine distance
       mt=pw.pairwise_distances(ftr1,ftr2,metric='cosine')
       predicts=np.empty((test_num,))
       for i in range(test_num):
              predicts[i]=mt[i][i]
       for i in range(test_num):
              predicts[i]=(predicts[i]-np.min(predicts))/(np.max(predicts)-np.min(predicts))

       print 'accuracy is:',accuracy(predicts,labels,test_num)

       np.savetxt('predicts.txt',predicts)
       fpr,tpr,thresholds=sklearn.metrics.roc_curve(labels,predicts)
       draw_roc(fpr,tpr,title=metric,save_name='lfw'+str(itera))

def accuracy(distance,labels,num):
       accuracy=[]
       predict=np.empty((num,))
       threshold=0.2
       while threshold<=0.8:
              for i in range(num):
                     if distance[i]>=threshold:
                            predict[i]=1
                     else:
                            predict[i]=0
              predict_right=0.0
              for i in range(num):
                     if predict[i]==labels[i]:
                            predict_right=1.0+predict_right
              current_accuracy=(predict_right/num)
              accuracy.append(current_accuracy)
              threshold=threshold+0.001
       return np.max(accuracy) 
```

The accuracy is always 0.5, 
and the predicts are abnormal ,for example:
...
1.727013593861347891e-15
1.727013593861347891e-15
2.072416312633617391e-15
3.108624468950426086e-15
3.108624468950426086e-15
1.727013593861347891e-15
...
And the ROC curve is not a curve，but a straight line...

I know something is wrong(the feature of extracking??) with the changed code.
Could you help me to check the test code?
Thank you very much!

**Comment by AlfredXiangWu** [comment]
The caffe-rc2 version has changed the interface. 
I recommend you to use caffe-rc python interface to extract features via my code, or you could use the new version python code(https://github.com/AlfredXiangWu/python_misc/pull/2) provided by @rmekdma.
